### PR TITLE
make kubectl config set-cluster easier to use

### PR DESCRIPTION
**Pull request by deads2k**
Addresses #3728 for set-cluster only.

@jlowdermilk If you agree with the approach, I'll update the other kubectl config set-\* commands.

### CODE REVIEW

File: docs/kubectl.md [hunk 1]
```diff
@@ -450,9 +450,9 @@ Usage:
```
Comment (jlowdermilk) [inline docs/kubectl.md]: Line [449](https://github.com/GoogleCloudPlatform/kubernetes/pull/3768/files#diff-2cd920f71449fac2cbb59c9daecd0aa4R449) is the one that needs to be changed. Usage string should list server|insecure-skip-tls-verify|certificate-authority|api-version as flags.

Comment (deads2k) [inline docs/kubectl.md]: > Line 449 is the one that needs to be changed.

Sorry, two branches and I missed that part of the change.

How about the new flag types for .Provided() and merging the values?

File: pkg/kubectl/cmd/config/create_cluster.go [hunk 2]
```diff
@@ -58,10 +59,12 @@ func NewCmdConfigSetCluster(out io.Writer, pathOptions *pathOptions) *cobra.Comm
...
-	cmd.Flags().StringVar(&options.server, clientcmd.FlagAPIServer, "", clientcmd.FlagAPIServer+" for the cluster entry in .kubeconfig")
-	cmd.Flags().StringVar(&options.apiVersion, clientcmd.FlagAPIVersion, "", clientcmd.FlagAPIVersion+" for the cluster entry in .kubeconfig")
-	cmd.Flags().BoolVar(&options.insecureSkipTLSVerify, clientcmd.FlagInsecure, false, clientcmd.FlagInsecure+" for the cluster entry in .kubeconfig")
-	cmd.Flags().StringVar(&options.certificateAuthority, clientcmd.FlagCAFile, "", clientcmd.FlagCAFile+" for the cluster entry in .kubeconfig")
+	options.insecureSkipTLSVerify.Default(false)
+
+	cmd.Flags().Var(&options.server, clientcmd.FlagAPIServer, clientcmd.FlagAPIServer+" for the cluster entry in .kubeconfig")
+	cmd.Flags().Var(&options.apiVersion, clientcmd.FlagAPIVersion, clientcmd.FlagAPIVersion+" for the cluster entry in .kubeconfig")
+	cmd.Flags().Var(&options.insecureSkipTLSVerify, clientcmd.FlagInsecure, clientcmd.FlagInsecure+" for the cluster entry in .kubeconfig")
+	cmd.Flags().Var(&options.certificateAuthority, clientcmd.FlagCAFile, clientcmd.FlagCAFile+" for the cluster entry in .kubeconfig")
```
Comment (jlowdermilk) [inline pkg/kubectl/cmd/config/create_cluster.go]: Are you sure this does what you want it to? I believe Cobra will call Set() on the variable with the default value if none is provided.

Comment (deads2k) [inline pkg/kubectl/cmd/config/create_cluster.go]: > Are you sure this does what you want it to?

Yes.  We use this technique in openshift: https://github.com/openshift/origin/blob/master/pkg/cmd/flagtypes/addr.go#L111.  pflag.Var looks clean: https://github.com/spf13/pflag/blob/master/flag.go#L355  
and https://gist.github.com/deads2k/53a9bd02412fe85f92e8

Comment (jlowdermilk) [inline pkg/kubectl/cmd/config/create_cluster.go]: Ok then, this approach with .Provided() lgtm

Comment (deads2k) [inline pkg/kubectl/cmd/config/create_cluster.go]: > this approach with .Provided() lgtm

Great.  I'll get on updating the other sets.

**Comment by jlowdermilk** [comment]
Approach looks good, go ahead and update other commands. While you're at it, can we get the same usage string for the other commands as well (usage string includes relevant --flags)?

**Comment by deads2k** [comment]
> can we get the same usage string for the other commands as well (usage string includes relevant --flags)?

Yes.  That was just a bad merge between branches (I tried two ways of checking .Provided()).  I will be sure to get the others.

**Comment by deads2k** [comment]
Update set-credentials, set-cluster, and set-context.  Here is an example of usage: https://gist.github.com/deads2k/466f31ba8a9824e14076

**Comment by jlowdermilk** [comment]
LGTM

**Comment by jlowdermilk** [comment]
Can you update the commit message to note that this changes all set-* commands? Happy to merge once that's accurate.

**Comment by deads2k** [comment]
@jlowdermilk  Done, but I think travis lost its mind.  Unrelated unit tests failed on 1.3 and passed on 1.4.  Can you kick it?

**Comment by jlowdermilk** [comment]
Rerunning travis.

**Comment by jlowdermilk** [comment]
All good. Merging.

**Pull request merged**